src/zarr/metadata.py

```diff
@@ -174,9 +175,9 @@ def __init__(
         chunk_grid: dict[str, JSON] | ChunkGrid,
         chunk_key_encoding: dict[str, JSON] | ChunkKeyEncoding,
         fill_value: Any,
-        codecs: Iterable[Codec | JSON],
-        attributes: None | dict[str, JSON],
-        dimension_names: None | Iterable[str],
+        codecs: Iterable[Codec | JSON] = (BytesCodec(),),
```

Comment:
I added some default values here to make array creation easier. happy to revert if this is controversial.